plugin/pkg/scheduler/schedulercache/node_info_test.go

```diff
@@ -0,0 +1,253 @@
+	}
+}
+
+func TestTaints(t *testing.T) {
+	node := makenode("test", nodelabels)
+	nodeinfo := makenodeinfo(node, taints)
+	gottaints, err := nodeinfo.Taints()
+	if err != nil {
+		t.Fatalf("get nodeinfo taints error: %s", err)
+	}
+	if !reflect.DeepEqual(gottaints, taints) {
+		t.Fatalf("get nodeinfo taints error,expected: %v,got: %v", taints, gottaints)
+	}
+}
+
+func TestMemoryPressureCondition(t *testing.T) {
+	node := makenode("test", nodelabels)
+	nodeinfo := makenodeinfo(node, taints)
+	nodeinfo.memoryPressureCondition = v1.ConditionTrue
+	if !reflect.DeepEqual(nodeinfo.MemoryPressureCondition(), nodeinfo.memoryPressureCondition) {
+		t.Fatalf("get nodeinfo memoryPressureCondition error,expected: %v,got: %v", nodeinfo.memoryPressureCondition, nodeinfo.MemoryPressureCondition())
+	}
+}
+
+func TestDiskPressureCondition(t *testing.T) {
+	node := makenode("test", nodelabels)
+	nodeinfo := makenodeinfo(node, taints)
+	nodeinfo.diskPressureCondition = v1.ConditionTrue
+	if !reflect.DeepEqual(nodeinfo.DiskPressureCondition(), nodeinfo.diskPressureCondition) {
+		t.Fatalf("get nodeinfo diskPressureCondition error,expected: %v,got: %v", nodeinfo.diskPressureCondition, nodeinfo.DiskPressureCondition())
+	}
+}
+
+func TestNonZeroRequest(t *testing.T) {
+	node := makenode("test", nodelabels)
+	nodeinfo := makenodeinfo(node, taints)
+	nodeinfo.nonzeroRequest = &testresource
+	if !reflect.DeepEqual(nodeinfo.NonZeroRequest(), *nodeinfo.nonzeroRequest) {
+		t.Fatalf("get nodeinfo diskPressureCondition error,expected: %v,got: %v", *nodeinfo.nonzeroRequest, nodeinfo.NonZeroRequest())
+	}
+}
+func TestAllocatableResource(t *testing.T) {
+	node := makenode("test", nodelabels)
+	nodeinfo := makenodeinfo(node, taints)
+	nodeinfo.allocatableResource = &testresource
+	if !reflect.DeepEqual(nodeinfo.AllocatableResource(), *nodeinfo.allocatableResource) {
+		t.Fatalf("get nodeinfo allocatableResource error,expected: %v,got: %v", *nodeinfo.allocatableResource, nodeinfo.AllocatableResource())
+	}
+}
+
+func TestClone(t *testing.T) {
+	node := makenode("test", nodelabels)
+	nodeinfo := makenodeinfo(node, taints)
+	niclone := nodeinfo.Clone()
+	if !reflect.DeepEqual(*niclone, nodeinfo) {
+		t.Fatalf("nodeinfo clone error,expected: %v,got: %v", nodeinfo, *niclone)
+	}
+}
+
+func TesthasPodAffinityConstraints(t *testing.T) {
+	podwithaffinity := makepod("foo", "", &affnity)
+	podwithoutaffinity := makepod("foo", "", &v1.Affinity{})
+	if !reflect.DeepEqual(hasPodAffinityConstraints(podwithaffinity), true) {
+		t.Fatalf("nodeinfo check pod has podaffinityconstraints error,expected: %v,got: %v", true, hasPodAffinityConstraints(podwithaffinity))
+	}
+	if !reflect.DeepEqual(hasPodAffinityConstraints(podwithoutaffinity), false) {
+		t.Fatalf("nodeinfo check pod has no podaffinityconstraints error,expected: %v,got: %v", false, hasPodAffinityConstraints(podwithoutaffinity))
+	}
+}
+
+func TestaddPod(t *testing.T) {

```

Comment:
I would probably collapse the tests add + set + remove into a single test that verifies in that order, as it's a more realistic use case.